src/armature/Model.ts

```diff
@@ -169,4 +174,94 @@ export class Model {
+        const materials: BakedMaterial[] = [];
+
+        this.computeRenderInfo(false).geometry.forEach((r: RenderObject) => {
+            // Give each material a unique number representing it. If a material has already been
+            // given a number, don't add it again, ensuring there are no duplicates.
+            if (!materialIndices.has(r.geometry.material)) {
+                materialIndices.set(r.geometry.material, materials.length);
+                materials.push(r.geometry.material);
+            }
+
+            // Add the vertex indices to a group with the specified material
+            groups.push({
+                // Use an array literal and ... to convert the int16 array to doubles
+                indices: chunk(
+                    [...r.geometry.indices].map((i: number) => i + vertices.length + 1),
+                    3
+                ),
+                material: <number>materialIndices.get(r.geometry.material)
+            });
+
+            // Add vertex and normals, transformed into world space
+            vertices.push(
+                ...chunk(r.geometry.vertices, 3).map(([x, y, z]: number[]) =>
+                    vec4.transformMat4(vec4.create(), vec4.fromValues(x, y, z, 1), r.transform)
+                )
+            );
+            normals.push(
+                ...chunk(r.geometry.normals, 3).map(([x, y, z]: number[]) =>
+                    vec3.transformMat3(vec3.create(), vec3.fromValues(x, y, z), r.normalTransform)
+                )
+            );
+        });
+
+        const obj = [
+            'o calderExport',
+
+            // Source the corresponding material file
+            'mtllib calderExport.mtl',
+
+            // List all vertices and normals
+            ...vertices.map(
+                (v: vec4) => `v ${v[0].toFixed(4)} ${v[1].toFixed(4)} ${v[2].toFixed(4)}`
+            ),
+            ...normals.map(
+                (v: vec3) => `vn ${v[0].toFixed(4)} ${v[1].toFixed(4)} ${v[2].toFixed(4)}`
+            ),
+
+            // Turn on smooth shading
+            's 1',
+
+            // Create a group for each distinct object, with the proper material applied
+            ...flatMap(groups, (g: { indices: number[][]; material: number }, i: number) => [
+                `g group${i}`,
+                `usemtl material${g.material}`,
+                ...g.indices.map((idx: number[]) => `f ${idx[0]} ${idx[1]} ${idx[2]}`)
+            ])
+        ].join('\n');
+
+        const ambient = ambientLightColor.asVec();
+        const mtl = flatMap(materials, (m: BakedMaterial, i: number) => [
+            `newmtl material${i}`,
+            `Ka ${ambient[0].toFixed(4)} ${ambient[1].toFixed(4)} ${ambient[2].toFixed(4)}`,
+
+            // In our shading model, diffuse and specular color are the same
+            `Kd ${m.materialColor[0].toFixed(4)} ${m.materialColor[1].toFixed(
+                4
```

Comment:
apologies for the weird autoformatter stuff here. If you prefer I can use normal strings concatenated with `+` instead of using `\``-quoted string interpolation so that lines can be broken up in less awkward ways